packages/core/src/crypto/BbsService.ts

```diff
@@ -0,0 +1,151 @@
+import type { CreateKeyOptions } from '../wallet'
+import type { BlsKeyPair as _BlsKeyPair } from '@mattrglobal/bbs-signatures'
+
+import {
+  bls12381toBbs,
+  generateBls12381G2KeyPair,
+  generateBls12381G1KeyPair,
+  sign,
+  verify,
+} from '@mattrglobal/bbs-signatures'
+
+import { TypedArrayEncoder } from '../utils/TypedArrayEncoder'
+import { Buffer } from '../utils/buffer'
+import { WalletError } from '../wallet/error'
+
+import { KeyType } from './KeyType'
+
+export interface BlsKeyPair {
+  publicKeyBase58: string
+  privateKeyBase58: string
+  keyType: Exclude<KeyType, KeyType.Ed25519 | KeyType.X25519>
+}
+
+interface BbsCreateKeyOptions extends CreateKeyOptions {
+  keyType: Exclude<KeyType, KeyType.X25519 | KeyType.Ed25519 | KeyType.Bls12381g1g2>
```

Comment:
Maybe we can invert this type so it still works if new key types are added?

```suggestion
  keyType: Pick<KeyType, KeyType.Bls12381g1 | KeyType.Bls12381g2>
```